includes/Registry/Registry.php

```diff
@@ -59,12 +61,89 @@ public function OnInit() {
 		UnknownBlock::register_type();
 	}
 
+	/**
+	 * Given the name of a block, return a list of Interfaces the block should implement to represent which contexts the block can be accessed from.
+	 *
+	 * i.e.
+	 * - a Block that is allowed on the "Post" post type will implement the "PostEditorBlock" interface
+	 * - a Block allowed on the "Page" post type will implement the "PageEditorBlock" interface
+	 * - a Block that is exposed in another context, for example NavMenu would implement the "NavMenuEditorBlock" interface (not yet supported)
+	 *
+	 * @param string $block_name The name of the block to get the interfaces for.
+	 *
+	 * @return array
+	 */
+	public function get_block_context_interfaces( string $block_name ): array {
+
+		// If there's already block interfaces defined for the block, return them
+		if ( ! empty( $this->block_interfaces[ $block_name ] ) ) {
+			return $this->block_interfaces[ $block_name ] ?? array();
+		}
+
+		// Get a list of Gutenberg and GraphQL enabled post types
+		$block_and_graphql_enabled_post_types = WPHelpers::get_supported_post_types();
+
+		if ( empty( $block_and_graphql_enabled_post_types ) ) {
+			return array();
+		}
+
+		$post_id = -1;
+
+		foreach ( $block_and_graphql_enabled_post_types as $post_type ) {
+			$type_name = strtolower( $post_type );
+
+			// retrieve a block_editor_context for the current post type
+			$block_editor_context = WPHelpers::get_block_editor_context( $post_type, $post_id-- );
+
+			// Fetch the list of allowed blocks for the current post type
+			$supported_blocks_for_post_type_context = get_allowed_block_types( $block_editor_context );
+
+			// If all blocks are supported in this context, we need a list of all the blocks
+			if ( true === $supported_blocks_for_post_type_context ) {
+				$supported_blocks_for_post_type_context = \WP_Block_Type_Registry::get_instance()->get_all_registered();
+				$supported_blocks_for_post_type_context = array_keys( $supported_blocks_for_post_type_context );
+			}
+
+			// If there are no supported blocks, return an empty array
+			if ( empty( $supported_blocks_for_post_type_context ) ) {
+				return array();
+			}
+
+			// Push the Contextual Interface for each supported Block to the block_interfaces array
+			array_map(
+				function( $supported_block ) use ( $type_name ) {
+					$this->block_interfaces[ $supported_block ][] = Utils::format_type_name( $type_name . 'EditorBlock' );
+				},
+				$supported_blocks_for_post_type_context
+			);
+		}//end foreach
+
+		return $this->block_interfaces[ $this->block->name ] ?? array();
```

Comment:
This line breaks tests since `$this->block->name` is not defined. Is this supposed to be `$block_name` instead?